File: Core/src/main/java/dev/lynith/core/utils/FileUtils.java
package dev.lynith.core.utils;

import dev.lynith.core.ClientStartup;

import java.nio.file.Path;
import java.nio.file.Paths;

public class FileUtils {

    public final static Path MC_DIR = Paths.get(ClientStartup.getInstance().getBridge().getGame().getGameDir().getAbsolutePath());
    public final static Path CONFIG_PATH = Paths.get(MC_DIR.toString(), "config", "zabu");

}

File: Core/src/main/kotlin/dev/lynith/core/utils/FileUtils.kt
package dev.lynith.core.utils

import dev.lynith.core.Platform
import java.io.File

object FileUtils {
    val GAME_DIR: File
        get() {
            val path = System.getProperty("zabu.gameDir", Platform.bridge.minecraft.gameDirectory)
            val file = File(path)

            if (!file.exists()) {
                file.mkdirs()
            }
            return file
        }

    val ZABU_FOLDER: File
        get() {
            val file = File(GAME_DIR, "zabu")
            if (!file.exists()) {
                file.mkdirs()
            }
            return file
        }

    val CONFIG_FOLDER: File
        get() {
            val file = File(ZABU_FOLDER, "config")
            if (!file.exists()) {
                file.mkdirs()
            }
            return file
        }
}
